### PR TITLE
Get orders

### PR DESCRIPTION
**Pull request by simionescuraluca**

### CODE REVIEW

File: src/main/java/com/vinyl/service/UserService.java [hunk 1]
```diff
@@ -203,6 +203,28 @@ public CustomerListDTO getAllCustomers(String tokenHash) {
...
+
+        User user = userRepository.findById(userId).orElseThrow(() -> new BadRequestException("User not found!"));
+
+        List<Purchase> purchases = purchaseRepository.findByUser(user);
+
+        List<OrderDTO> customerOrders = new ArrayList<>();
+
+        for (Purchase p : purchases) {
+            OrderDTO order = new OrderDTO();
+            Double orderCost = p.getProducts().stream().reduce(0.0, (a, ab) -> a + (ab.getNrItems() * ab.getProduct().getPrice()), (aa, bb) -> null);
```
Comment (dragosslujeru) [inline src/main/java/com/vinyl/service/UserService.java]: do this in a separate method

Comment (simionescuraluca) [reply src/main/java/com/vinyl/service/UserService.java]: done

File: src/test/java/com/vinyl/GetAllCustomerOrdersTest.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+        product = defaultEntitiesHelper.createProduct();
+        order = purchaseSetup();
+    }
+
+    @Test
+    public void testWhenOk() {
+        ResponseEntity<OrderListDTO> response = setUpHeaderAndGetTheResponse();
+        Assertions.assertThat(response.getStatusCode()).isEqualTo(OK);
+        Assertions.assertThat(response.getBody().getOrders().size()).isEqualTo(1);
+        Assertions.assertThat(response.getBody().getOrders().get(0).getCost()).isEqualTo(product.getPrice() * purchaseProductRepository.findByPurchase(order).getNrItems());
```
Comment (simionescuraluca) [inline src/test/java/com/vinyl/GetAllCustomerOrdersTest.java]: use absolute value

Comment (simionescuraluca) [inline src/test/java/com/vinyl/GetAllCustomerOrdersTest.java]: see floating point comparison

File: src/main/java/com/vinyl/controller/UserController.java [hunk 3]
```diff
@@ -138,4 +138,21 @@
...
+    @ApiOperation(value = "Admin gets all orders of a customer", response = ResponseEntity.class)
+    @ApiResponses(value = {
+            @ApiResponse(code = 401, message = NOT_AUTHORIZED_MESSAGE),
+            @ApiResponse(code = 400, message = BAD_REQUEST_MESSAGE),
+            @ApiResponse(code = 200, message = "You successfully got the orders")
+
+    })
+    @RequestMapping(value = "/users/{userId}/orders", method = RequestMethod.GET)
+    public ResponseEntity<?> getAllCustomerOrders(@ApiParam(value = "Token hash to send in the request header", required = true) @RequestHeader(value = "Authorization", required = false) String auth,
+                                                  @ApiParam(value = "Token hash to send in the request header", required = true) @PathVariable Integer userId) {
```
Comment (dragosslujeru) [inline src/main/java/com/vinyl/controller/UserController.java]: User ID, not hash and is not sent in header

Comment (simionescuraluca) [reply src/main/java/com/vinyl/controller/UserController.java]: fixed

File: src/test/java/com/vinyl/GetAllCustomerOrdersTest.java (hunk 2 above)
Comment (simionescuraluca) [reply src/test/java/com/vinyl/GetAllCustomerOrdersTest.java]: done

Comment (simionescuraluca) [reply src/test/java/com/vinyl/GetAllCustomerOrdersTest.java]: done

File: src/test/java/com/vinyl/GetAllCustomerOrdersTest.java [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+        Assertions.assertThat(response.getBody().getOrders().get(0).getCost()).isEqualTo(expectedCost);
+    }
+
+    @Test
+    public void testWhenUserIdIsInvalid() {
+        ResponseEntity<?> response = setUpHeaderAndGetTheResponse("/users/" + 00 + "/orders");
+        Assertions.assertThat(response.getStatusCode()).isEqualTo(HttpStatus.BAD_REQUEST);
+    }
+
+    protected ResponseEntity<OrderListDTO> setUpHeaderAndGetTheResponse() {
```
Comment (simionescuraluca) [inline src/test/java/com/vinyl/GetAllCustomerOrdersTest.java:57]: override

Comment (simionescuraluca) [reply src/test/java/com/vinyl/GetAllCustomerOrdersTest.java:57]: done

**Review by dragosslujeru — APPROVED** [review]

**Pull request merged**